### PR TITLE
Support proxy attribute for download

### PR DESCRIPTION
**Pull request by koraniar**
This PR can be helpful for #490.  You will not need to copy the proxy settings to the system account (that can cause errors in other applications); just modify the XML configuration file (manually or programmatically) and restart the service.

### CODE REVIEW

File: doc/xmlConfigFile.md [hunk 1]
```diff
@@ -186,19 +186,23 @@ For target servers using the HTTPS transfer protocol it is necessary, that the C
 By default, the `download` command does not fail the service startup if the operation fails (e.g. `from` is not available).
 In order to force the download failure in such case, it is possible to specify the `failOnError` boolean attribute.
 
+To specify a custom proxy you can put your proxy URL in the parameter `proxyServer`.
```
Comment (NextTurn) [inline doc/xmlConfigFile.md:189]:
```suggestion
To specify a custom proxy you can put your proxy address in the parameter `proxyServer`.
```

File: doc/xmlConfigFile.md [hunk 2]
```diff
@@ -186,19 +186,23 @@ For target servers using the HTTPS transfer protocol it is necessary, that the C
...
 
+<download from="http://example.com/some.dat" to="%BASE%\some.dat" proxyServer="http://192.168.1.5:80/"/>
+
 <download from="https://example.com/some.dat" to="%BASE%\some.dat" auth="sspi" />
 
 <download from="https://example.com/some.dat" to="%BASE%\some.dat" failOnError="true"
           auth="basic" user="aUser" password="aPassw0rd" />
 
-<download from="http://example.com/some.dat" to="%BASE%\some.dat"
+<download from="http://example.com/some.dat" to="%BASE%\some.dat" proxyServer="http://192.168.1.5:80/"
```
Comment (NextTurn) [inline doc/xmlConfigFile.md:205]: I don't like the long name.

```suggestion
<download from="http://example.com/some.dat" to="%BASE%\some.dat" proxy="http://192.168.1.5:80/"
```

**Comment by NextTurn** [comment]
I'd guess an address in the form of `http://USERNAME:PASSWORD@HOST:PORT/` is sufficient to carry all information needed.

**Comment by koraniar** [comment]
> I'd guess an address in the form of `http://USERNAME:PASSWORD@HOST:PORT/` is sufficient to carry all information needed.

Mmm, no it will not work with credentials, I am going to update the PR to use the credentials in the format you proposed so as not to create more attributes 🙂.

**Comment by koraniar** [comment]
@oleg-nenashev I moved the logic of extract the credentials from the proxy address to an independent class in the same `Download.sc` file to add the Unit Tests.

**Comment by ricardopolo** [comment]
Please merge it!!! :shipit:

**Comment by ricardopolo** [comment]
@oleg-nenashev @NextTurn would you mind releasing a new version, could be a minor, please?

**Comment by oleg-nenashev** [comment]
Hi. I will do it tomorrow or on the weekend. The new release is already
overdue

**Comment by oleg-nenashev** [comment]
@ricardopolo @koraniar 2.9.0 should be out now: https://github.com/winsw/winsw/releases/tag/v2.9.0 . Please let us know if you notice any issues in the new release

**Pull request merged**